File: 백준/Bronze/1978. 소수 찾기/소수 찾기.py
import math

count = 0

n = input()
nlist = list(map(int,input().split()))

for i in nlist:
    flag = 0
    for j in range(2,int(math.sqrt(i)+1)):
        if i%j==0:
            flag = 1
            break
    if i==1:
        flag = 1
    if flag==0:
        count+=1
print(count)

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    let answer = [];
    let arr = numbers.split("");
    let count = 0;
    const checkSOSU = (num) =>{
        if(num<2)
            return false
        for(let i=2;i<=Math.sqrt(num);i++)
            if(num%i===0)
                return false
        return true
    }
    const DFS = (index,num) =>{
        for(let i=0;i<numbers.length;i++){
            if(!(index.includes(i))){
                index.push(i)
                num.push(numbers[i])
                let ele = Number(num.join(""))
                if(checkSOSU(ele)&&!(answer.includes(ele)))
                    answer.push(ele)
                DFS(index,num)
                num.pop();
                index.pop();
            }
        }
    }
    DFS([],[])
    console.log(answer)
    return answer.length;
}
//반복문으로 짜보기 재귀는 쉬움
